requirements.txt

```diff
@@ -52,15 +52,15 @@ python-dateutil==2.9.0.post0 ; python_version >= "3.11" and python_version < "3.
 python-string-utils==1.0.0 ; python_version >= "3.11" and python_version < "3.13"
 pyvmomi==8.0.2.0.1 ; python_version >= "3.11" and python_version < "3.13"
 pyyaml==6.0.1 ; python_version >= "3.11" and python_version < "3.13"
-redis==5.0.4 ; python_version >= "3.11" and python_version < "3.13"
+redis==5.0.5 ; python_version >= "3.11" and python_version < "3.13"
 requests-oauthlib==2.0.0 ; python_version >= "3.11" and python_version < "3.13"
-requests==2.32.2 ; python_version >= "3.11" and python_version < "3.13"
+requests==2.32.3 ; python_version >= "3.11" and python_version < "3.13"
 resolvelib==1.0.1 ; python_version >= "3.11" and python_version < "3.13"
 rsa==4.9 ; python_version >= "3.11" and python_version < "3.13"
-setuptools==69.5.1 ; python_version >= "3.11" and python_version < "3.13"
+setuptools==70.0.0 ; python_version >= "3.11" and python_version < "3.13"
```

Comment:
downgrade[2]?

Reply:
IMHO I prefer to downgrade and keep setuptools in both files to the same version.

Reply:
This would mean converting setuptools from a indirect to direct dependency. Why? Just because to keep it in sync with transitive dependencies (which have no obligation to be in sync with actual runtime dependencies - they could all be in conflict and it would be fine).

If you are installing `requirements-build.txt`, you are doing something wrong. This file only exists to do dependency resolution of transitive dependencies that will be used downstream. It could have multiple versions of setuptools if required and it would be fine.